src/DIPS.Xamarin.UI/Controls/FloatingActionMenu/FloatingActionMenu.xaml.cs

```diff
@@ -0,0 +1,307 @@
+        {
+            get => (Color)GetValue(ExpandButtonBackgroundColorProperty);
+            set => SetValue(ExpandButtonBackgroundColorProperty, value);
+        }
+
+        /// <summary>
+        /// </summary>
+        public string ExpandButtonText
+        {
+            get => (string)GetValue(ExpandButtonTextProperty);
+            set => SetValue(ExpandButtonTextProperty, value);
+        }
+
+        /// <summary>
+        ///     FontFamily of the text in the expand button.
+        /// </summary>
+        public string FontFamily
+        {
+            get => (string)GetValue(FontFamilyProperty);
+            set => SetValue(FontFamilyProperty, value);
+        }
+
+        /// <summary>
+        /// </summary>
+        public double Size
+        {
+            get => (double)GetValue(SizeProperty);
+            set => SetValue(SizeProperty, value);
+        }
+
+        /// <summary>
+        ///     Set to true to close the menu if its open.
+        /// </summary>
+        public bool CloseMenu
+        {
+            get => (bool)GetValue(CloseMenuProperty);
+            set => SetValue(CloseMenuProperty, value);
+        }
+
+        private void Hide()
+        {
+            if (m_animationComplete) AnimateAll(true);
+        }
+
+        private static void CloseMenuPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
+        {
+            if (bindable is FloatingActionMenu menu && newvalue is true)
+            {
+                menu.Hide();
+                menu.CloseMenu = false;
+            }
+        }
+
+        private async Task AnimateAll(bool isExpanded)
+        {
+            m_animationComplete = false;
+
+            if (!isExpanded)
+            {
+                m_overLay.IsVisible = true;
+                m_overLay.FadeTo(.5);
+            }
+            else
+            {
+                m_overLay.FadeTo(0);
+            }
+
+            var multiplier = 1;
+
+            foreach (var menuButton in Children)
+            {
+                menuButton.InputTransparent = isExpanded;
+                var maxOpacity = menuButton.IsEnabled ? 1 : .5;
+                menuButton.TranslateTo(0, isExpanded ? 0 : -m_yTranslate * multiplier, 250, Easing.CubicInOut);
+                menuButton.FadeTo(isExpanded ? 0 : maxOpacity, 250, Easing.CubicInOut);
+                multiplier += 1;
+            }
+
+            if (isExpanded) ExpandButton.FadeTo(.5, 250, Easing.CubicInOut);
+            else ExpandButton.FadeTo(1, 250, Easing.CubicInOut);
+
+            await ExpandButton.RelRotateTo(180, 250, Easing.CubicInOut);
+            if (isExpanded) m_overLay.IsVisible = false;
+            m_isExpanded = !isExpanded;
+            m_animationComplete = true;
+        }
+
+        private async void ExpandButton_OnClicked(object sender, EventArgs e)
+        {
+            if (m_first) AdjustXPositions();
+            if (m_animationComplete) await AnimateAll(m_isExpanded);
```

Comment:
Just set ExpandButton.InputTransparent = true before and false after Animate all.

Reply:
Doesn't work if the expand button is clicked quickly enough. The change of input transparancy is apparently not instant